src/command_modules/azure-cli-ml/azure/cli/command_modules/ml/env.py

```diff
@@ -405,14 +407,11 @@ def env_setup(status, name, kubernetes, local_only, context=CommandLineInterface
     else:
         root_name = name
 
-    try:
-        az_login()
-        if not name:
-            az_check_subscription()
-        resource_group = az_create_resource_group(context, root_name)
-    except AzureCliError as exc:
-        print(exc.message)
-        return
+    if service_principal_app_id and not service_principal_password:
+        raise AzureCliError('When deploying with service principal, client secret must be specified.')
+    az_login()
```

Comment:
None of the functions raise an AzureCliError. I have updated the AzureCliError raised at line 411 to be a CLIError, which is handled by azure-cli main.